### PR TITLE
First batch: Replace `MatrixClient.isRoomEncrypted` by `MatrixClient.CryptoApi.isEncryptionEnabledInRoom`

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/element-web)

Task https://github.com/element-hq/element-web/issues/26922
Replace `MatrixClient.isRoomEncrypted` by `MatrixClient.CryptoApi.isEncryptionEnabledInRoom`

### CODE REVIEW

File: src/utils/crypto/shouldSkipSetupEncryption.ts [hunk 1]
```diff
@@ -9,14 +9,21 @@ Please see LICENSE files in the repository root for full details.
...
-export const shouldSkipSetupEncryption = (client: MatrixClient): boolean => {
+export const shouldSkipSetupEncryption = async (client: MatrixClient): Promise<boolean> => {
     const isEncryptionForceDisabled = shouldForceDisableEncryption(client);
-    return isEncryptionForceDisabled && !client.getRooms().some((r) => client.isRoomEncrypted(r.roomId));
+    const crypto = client.getCrypto();
+    if (!crypto) return true;
+
+    return (
+        isEncryptionForceDisabled &&
+        !(await asyncSome(client.getRooms(), ({ roomId }) => crypto.isEncryptionEnabledInRoom(roomId)))
```
Comment (t3chguy) [inline src/utils/crypto/shouldSkipSetupEncryption.ts:27]: This seems unnecessarily slow, should it not be a Promise.race?

File: src/Searching.ts [hunk 2]
```diff
@@ -605,7 +605,7 @@ function eventIndexSearch(
     let searchPromise: Promise<ISearchResults>;
 
     if (roomId !== undefined) {
-        if (client.isRoomEncrypted(roomId)) {
+        if (client.getCrypto()?.isEncryptionEnabledInRoom(roomId)) {
```
Comment (t3chguy) [inline src/Searching.ts:608]: this if statement will always be truthy as a Promise is truthy, it should be awaited

File: src/DeviceListener.ts [hunk 3]
```diff
@@ -240,13 +241,16 @@ export default class DeviceListener {
...
         // If we're in the middle of a secret storage operation, we're likely
         // modifying the state involved here, so don't add new toasts to setup.
         if (isSecretStorageBeingAccessed()) return false;
         // Show setup toasts once the user is in at least one encrypted room.
         const cli = this.client;
-        return cli?.getRooms().some((r) => cli.isRoomEncrypted(r.roomId)) ?? false;
+        const cryptoApi = cli?.getCrypto();
+        if (!cli || !cryptoApi) return false;
+
+        return asyncSome(cli.getRooms(), ({ roomId }) => cryptoApi.isEncryptionEnabledInRoom(roomId));
```
Comment (t3chguy) [inline src/DeviceListener.ts:253]: Same comment as elsewhere, should we not be using Promise.race

File: src/utils/crypto/shouldSkipSetupEncryption.ts (hunk 1 above)
Comment (florianduros) [reply src/utils/crypto/shouldSkipSetupEncryption.ts:27]: I don't think so. `Promise.race` fulfils the first promise to settle which is not good for this scenario:
1. Two rooms, two promises: p1 & p2
2. If p1 settles first and the room is unencrypted => `Promise.race` returns `false`
3. If p2 settles first and the room is encrypted => `Promise.race` returns `true`

`asyncSome` stops to iterate when a promise is resolved.

Maybe thought about `Promise.any` and to reject the promises when the room is encrypted?

**Review by t3chguy — APPROVED** [review]

**Pull request merged**